### PR TITLE
Fix testMasterFailoverDuringCloneStep1

### LINKED ISSUE #63473: [CI] CloneSnapshotIT.testMasterFailoverDuringCloneStep1 failure
**Build scan**:
https://gradle-enterprise.elastic.co/s/a6lovmsnlvypu

**Repro line**:
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.snapshots.CloneSnapshotIT.testMasterFailoverDuringCloneStep1" \
  -Dtests.seed=79DAA05140C620AC \
  -Dtests.security.manager=true \
  -Dtests.locale=sq \
  -Dtests.timezone=Africa/Lagos \
  -Druntime.java=11

**Reproduces locally?**:
no

**Applicable branches**:
master

**Failure history**:
Two more since Oct 5th:
https://gradle-enterprise.elastic.co/s/lpt6ip6vz7sbc
https://gradle-enterprise.elastic.co/s/tpnuofyxi2ucq

**Failure excerpt**:
java.lang.AssertionError: Found leaked snapshot completion listeners {test-repo:target-snapshot/XrAiUKcgSkaoFGtCvH3XMg=[org.elasticsearch.action.ActionListener$1@64a0aff0]} on [{node_t2}{AR7VHs2TRf2LNBefAhu9vA}{xuFfSHmZTF-v9duV1ushWg}{127.0.0.1}{127.0.0.1:46509}{m}]
	at org.elasticsearch.snapshots.SnapshotsService.assertAllListenersResolved(SnapshotsService.java:2239)
	at org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase.lambda$verifyNoLeakedListeners$0(AbstractSnapshotIntegTestCase.java:123)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:937)

### PR DESCRIPTION
**Pull request by original-brownbear**
Assuming the clone failed when the request failed is not sufficient.
There are failure modes where the request fails but the clone still works out
because the data node resent the requeest after the first clone had already been
failed and removed from the cluster state when master was restarted.

Closes #63473

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java [hunk 1]
```diff
@@ -377,6 +378,9 @@ public void testMasterFailoverDuringCloneStep1() throws Exception {
 
         awaitNoMoreRunningOperations(internalCluster().getMasterName());
 
+        // Check if the clone operation worked out by chance as a result of the clone request being retried because of the master failover
+        cloneSucceeded = cloneSucceeded ||
```
Comment (original-brownbear) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java:382]: It's mildly annoying that we have to do this kind of thing again (we had this issue in tests with the snapshot `INIT` step as well). There is a reasonable fix here I think (I'll try it when I can find some time but probably not too soon) but for now I opted to just fix the test.

Comment (tlrx) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java:382]: Maybe add a //TODO or a descriptive comment somewhere in the code?

Comment (original-brownbear) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java:382]: Added :) in the relevant spots in 060020a

**Review by tlrx — APPROVED** [review]

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Pull request merged**